torch/csrc/jit/passes/quantization.cpp

```diff
@@ -637,16 +634,32 @@ void InsertQuantDeQuantHelper::removeObservers(script::Module& module) {
   }
 }
 
-void InsertQuantDeQuantHelper::quantizeTensors(script::Module& module) {
+void InsertQuantDeQuantHelper::quantizeTensors(script::Module& module, Value* self) {
   if (!values_to_quantize_.count(module._ivalue())) {
     return;
   }
   for (auto& v : values_to_quantize_.at(module._ivalue())) {
     TORCH_INTERNAL_ASSERT(values_to_qparams_.at(module._ivalue()).count(v));
-    auto tp = values_to_qparams_.at(module._ivalue()).at(v);
-    auto qparams = std::get<0>(tp);
-    auto scalar_type = std::get<1>(tp);
-    insertQuantDeQuantCall(v, qparams, scalar_type);
+    auto qparams_and_scalar_type = values_to_qparams_.at(module._ivalue()).at(v);
+    auto qparams = std::get<0>(qparams_and_scalar_type);
+    auto scalar_type = std::get<1>(qparams_and_scalar_type);
+    // Register attributes for quantization parameters
+    auto tp = qparams.toTuple();
+    at::Tensor scale = tp->elements()[0].toTensor().to(at::kFloat);
+    at::Tensor zero_point = tp->elements()[1].toTensor().to(at::kInt);
+    // TODO: get this info from qscheme
+    bool is_per_channel = scale.numel() > 1;
+    std::string prefix = v->debugName();
+    if (is_per_channel) {
+      module.register_attribute(prefix + "_scale", TensorType::get(), scale);
+      module.register_attribute(prefix + "_zero_point", TensorType::get(), zero_point);
+      module.register_attribute(prefix + "_axis", IntType::get(), tp->elements()[2].toInt());
+    } else {
+      module.register_attribute(prefix + "_scale", FloatType::get(), scale.item<double>());
+      module.register_attribute(prefix + "_zero_point", IntType::get(), zero_point.item<int64_t>());
+    }
+    module.register_attribute(prefix + "_scalar_type", IntType::get(), scalar_type);
```

Comment:
Could this code be moved inside `insertQuantDeQuantCall` or factored out to a separate function?